RecoCTPPS/TotemRPLocal/plugins/TotemTimingRecHitProducer.cc

```diff
@@ -59,6 +63,13 @@ TotemTimingRecHitProducer::produce( edm::Event& iEvent, const edm::EventSetup& i
 {
   std::unique_ptr<edm::DetSetVector<TotemTimingRecHit> > pOut( new edm::DetSetVector<TotemTimingRecHit> );
 
+  // check for timing calibration parameters update
+  if ( calibWatcher_.check( iSetup ) ) {
+    edm::ESHandle<PPSTimingCalibration> hTimingCalib;
+    iSetup.get<PPSTimingCalibrationRcd>().get( hTimingCalib );
+    algo_.setCalibration( *hTimingCalib );
```

Comment:
one possible optimization that could also currently remove the need to have the payloads for MC is to first check the digis in the event record. If it is empty, there is no need to load the ES products or even call the algo.

Reply:
@slava77 Indeed, this is one possible (and working as previously tested) workaround! But I found it personally more confusing (and possibly more prone to false negative on data for incorrect streams) to propagate this check for the validity of the various input collections in all producers downstream. What do you think?

Reply:
Implemented in f006eb14352eedfc4b9d725c36f413e9abfd4e8c